app/assets/javascripts/autocomplete.js

```diff
@@ -130,282 +141,296 @@ var MOAutocompleter = function(opts) {
+      window.addEventListener("beforeunload", function (event) {
+        return this2.our_unload()
+      });
     }
 
-    // Disable default browser autocomplete.
-    elem.attr("autocomplete", "off");
+    // Disable default browser autocomplete. Stimulus - do this on HTML element
+    elem.setAttribute("autocomplete", "off");
   },
 
-// ------------------------------ Events ------------------------------
+  // ------------------------------ Events ------------------------------
 
   // User pressed a key in the text field.
-  our_keydown: function(event) {
-    var key = event.which == 0 ? event.keyCode : event.which;
-    // jQuery("#log").append("keydown(" + key + ")<br>");
+  our_keydown: function (event) {
+    const key = event.which == 0 ? event.keyCode : event.which;
+    // this.debug("keydown(" + key + ")");
     this.clear_key();
     this.focused = true;
     if (this.menu_up) {
       switch (key) {
-      case EVENT_KEY_ESC:
-        this.schedule_hide();
-        this.menu_up = false;
-        break;
-      case EVENT_KEY_RETURN:
-      case EVENT_KEY_TAB:
-        if (this.current_row >= 0)
-          this.select_row(this.current_row - this.scroll_offset);
-        break;
-      case EVENT_KEY_HOME:
-        this.go_home();
-        break;
-      case EVENT_KEY_END:
-        this.go_end();
-        break;
-      case EVENT_KEY_PAGEUP:
-        this.page_up();
-        this.schedule_key(this.page_up);
-        break;
-      case EVENT_KEY_UP:
-        this.arrow_up();
-        this.schedule_key(this.arrow_up);
-        break;
-      case EVENT_KEY_DOWN:
-        this.arrow_down();
-        this.schedule_key(this.arrow_down);
-        break;
-      case EVENT_KEY_PAGEDOWN:
-        this.page_down();
-        this.schedule_key(this.page_down);
-        break;
-      default:
-        this.current_row = -1;
-        break;
+        case EVENT_KEY_ESC:
+          this.schedule_hide();
+          this.menu_up = false;
+          break;
+        case EVENT_KEY_RETURN:
+        case EVENT_KEY_TAB:
+          if (this.current_row >= 0)
+            this.select_row(this.current_row - this.scroll_offset);
+          break;
+        case EVENT_KEY_HOME:
+          this.go_home();
+          break;
+        case EVENT_KEY_END:
+          this.go_end();
+          break;
+        case EVENT_KEY_PAGEUP:
+          this.page_up();
+          this.schedule_key(this.page_up);
+          break;
+        case EVENT_KEY_UP:
+          this.arrow_up();
+          this.schedule_key(this.arrow_up);
+          break;
+        case EVENT_KEY_DOWN:
+          this.arrow_down();
+          this.schedule_key(this.arrow_down);
+          break;
+        case EVENT_KEY_PAGEDOWN:
+          this.page_down();
+          this.schedule_key(this.page_down);
+          break;
+        default:
+          this.current_row = -1;
+          break;
       }
     }
-    if (this.on_keydown)
-      this.on_keydown(event);
```

Comment:
@pellaea can you remember what you meant here? `on_keydown` is not a JS method, it's `onkeydown`. 

There's a series of overrides here but it's not clear to me what these statements do. `this.onkeydown` always fires an event.

```js
if (this.on_keydown)
      this.on_keydown(event);
```

Reply:
My best guess is that this originated in the template I started with.  We don't seem to use these callbacks anywhere, so feel free to tear them out.  (It's not just on_keydown, there's also an on_keypress and on_keyup.)  I think you can simply remove those 6 lines and that's it.

Reply:
Thank you. 
Do you have any time today to give this a spin and see if this rewrite works as intended?

 It's not quite ready for review but i will mark it as such if that's a possibility.

Reply:
Ok. I won't mark it for review unless i'm pretty sure it's bug free.

Reply:
Ready for review, but no rush, safe travels. 
jQuery extension methods are duplicated here in JS until they can be moved into a utility class.

Reply:
At the gate, and need to stay awake... holy moly, this is fast!  Ah, is this what the rest of the world is like???